usr/src/uts/common/fs/zfs/dsl_dir.c

```diff
@@ -940,11 +940,17 @@ dsl_dir_set_logicalquota_sync(void *arg, dmu_tx_t *tx)
 	VERIFY0(dsl_dataset_hold(dp, ddslqa->ddslqa_name, FTAG, &ds));
 	dd = ds->ds_dir;
 
-	dsl_prop_set_sync_impl(ds, zfs_prop_to_name(ZFS_PROP_LOGICALQUOTA),
-	    ddslqa->ddslqa_source, sizeof (ddslqa->ddslqa_value), 1,
-	    &ddslqa->ddslqa_value, tx);
-	VERIFY0(dsl_prop_get_int_ds(ds,
-	    zfs_prop_to_name(ZFS_PROP_LOGICALQUOTA), &newval));
+	if (spa_version(dp->dp_spa) >= SPA_VERSION_RECVD_PROPS) {
```

Comment:
It looks like if the pool version is older than `RECVD_PROPS`, this code is not setting the property, but is logging it.  Why do we want to do that?  If we can't support `logicalquota` on older pools, we should return `ENOTSUP`, and not log it.